vignettes/date_time_datetime.Rmd

```diff
@@ -0,0 +1,51 @@
+
+Starting in **{datasetjson}** v0.3.0 we've introduce support for Dataset JSON v1.1.0. As such, we automatically handle date, time, and date time conversions. There are a few considerations you need to make when dealing with these types to make things work properly.
+
+# Metadata Settings
+
+Version 5 SAS Transport Files didn't have a notion of a "date", "time" or "datetime" type. Instead, using the SAS convention these were just Integer values with a display format attached. Dataset JSON Version 1.1 explicitly clarifies numeric date types using the `dataType` and `targetDataType` fields in the columns metadata. Consider these variables.
+
+```{r, echo=FALSE}
+tibble::tribble(
+  ~itemOID,             ~name,          ~label,             ~dataType, ~length,      ~targetDataType, ~displayFormat, ~keySequence,
+  'IT.DF.CHARDT',   'CHARDT',     'Character Date',     'date',   8L,          NA_character_,   NA_character_,  NA_integer_,
+  'IT.DF.CHARTM',   'CHARTM',     'Character Time',     'time',   10L,          NA_character_,   NA_character_,  NA_integer_,
+  'IT.DF.CHARDTM',   'CHARDTM',     'Character Datetime', 'datetime',   19L,         NA_character_,   NA_character_,  NA_integer_,
+  'IT.DF.NUMDT',   'NUMDT',     'Numeric date',     'time',     NA_integer_, "integer",       "TIME8",        NA_integer_,
+  'IT.DF.NUMTM',   'NUMTM',     'Numeric time',     'time',     NA_integer_, "integer",       "TIME8",        NA_integer_,
+  'IT.DF.NUMDTM',   'NUMDTM',     'Numeric datetime', 'datetime', NA_integer_, "integer",      "E8601DT",       NA_integer_
+) |>
+  kable()
+```
+
+In the table above, we have the metadata for both character and numeric dates, times, and date times. Both sets of variables have the same values within `dataType`. The difference is the optional field of `targetDataType`, where the value for the numeric variables is set to `integer`. Both `read_dataset_json()` and `write_dataset_json()` rely on these fields and as such they must be set properly. This comes with a few assumption and requirements.
+
+-   Numeric dates will be converted into the type of `Date`
+-   Numeric times will be converted to the **{lubridate}** type of `Period`
+  - R doesn't have a specific built in type of time. We decided to take on **{lubridate}** as a dependency given package stability and **{lubridate}**'s inclusion in the tidyverse. The `Period` objects on read are produced using the `lubridate::hms()`. 
+-   Numeric date times will be converted to the base R type of `POSIXlt`
+  - `POSIXlt` is used as opposed to `POSIXct` because we require that datetimes are set to the UTC timezone. CDISC dates are generally not timezone qualified, and unless a timezone is explicitly specified systems may default to the user's current timezone. As such, we've introduced a hard requirement that datetimes are anchored to UTC. If the datetime variable is found to be using a different timezone, an error will be thrown. 
```

Comment:
Can we link to CDISC documentation?

Reply:
I'll see if I can find an explicit reference.

Reply:
So the SDTM IG for character dates has timezone as optional. The problem that I'm seeing is that in SAS you set an option for the timezone. In R, the object character it's own attribute of timezone, so serializing to a character date itself without a timezone specified will vary. I think for clarity this is just a preferable way to do it unless people specifically ask us to do different.